### DISCUSSION TITLE
How to use keys ]]

### DISCUSSION THREAD

**Discussion by AckslD** [wbthomason/packer.nvim#485] [Q&A]
I'm trying to lazy load `nvim-treesitter-textobjects` on keys where one is `]]`
```lua
use {
    'nvim-treesitter/nvim-treesitter-textobjects',
    module = 'nvim-treesitter-textobjects',
    keys = {
        {'o', 'af'},
        ...
        "]m",
        "]]",
        ...
}
```
but in `packer_complied.lua` I then get:
```lua
vim.cmd [[noremap <silent> ]] <cmd>lua require("packer.load")({'nvim-treesitter-textobjects'}, { keys = "[M", prefix = "" }, _G.packer_plugins)<cr>]]
```

How can I escape `]]` properly?

**Comment by wbthomason** [comment]
Ah, this is a hopefully temporary limitation on our end. `]]` is the sequence for "end raw string" in Lua, and we pass raw strings to `vim.cmd` to define keymaps. So, having that sequence inside the raw string terminates it prematurely and we get an error. Simply escaping it doesn't work either, because then we make a mapping to the literal string `\]\]` rather than what you wanted.

We could work around this by special-casing that keymap and adding a more complicated string constructor for the lazy-load keymap which breaks around the `]]` to allow it to be used, but this seems like a significant amount of work for a niche case.

More ideally, when either we get a better mechanism for making keymaps in Lua (coming soon, hypothetically) or, potentially, as a side effect of #402 (since we won't need the string output of the argument to `vim.cmd`), this will go away.

Sorry for the probably unsatisfying answer!

**Reply by AckslD** [reply]
Thanks for the clarification @wbthomason, I don't think this is of very high priority and sounds good that it can be fixed when refactoring :+1:

**Reply by wbthomason** [reply]
Noting to myself: there might be a way to fix this, because I've learned about the "Nesting Quotes" section of http://lua-users.org/wiki/StringsTutorial. I'll still need to figure out exactly what to modify, but this gives us a mechanism for allowing maps like this.
